privacyidea/static/components/token/controllers/tokenControllers.js

```diff
@@ -667,8 +671,14 @@ myApp.controller("tokenEnrollController", ["$scope", "TokenFactory", "$timeout",
                         token.subject
                             = (response.detail.u2fRegisterResponse || response.detail.webAuthnRegisterResponse).subject;
                         token.vendor = token.subject.split(" ")[0];
-                        console.log(token);
+                        //console.log(token);
+                    });
+            }, function (error) {
+                if (AuthFactory.checkRight("delete")) {
+                    TokenFactory.delete($scope.serial, function (response) {
```

Comment:
When do we get into the error function? When a user hits cancel?
Because i think we never deleted tokens through the API automatically. What happens if the user has no `DELETE` rights?

Reply:
Yes you get the error callback if any "error" occurs, like hitting cancel in any dialog. If the user has no delete right, the code is not executed and you get the shitty behavior from before where you are stuck on the "waiting for completion" page.